File: MS.NET/Fundamentals/Platform/Concurrency/threadtest.cs
using System;
using System.Threading;

static class Program
{
	private static void HandleJob(int jid)
	{
		Console.WriteLine("Thread<{0}> has accepted job:{1}", Thread.CurrentThread.ManagedThreadId, jid);
		Worker.DoWork(jid);
		Console.WriteLine("Thread<{0}> has finished job:{1}", Thread.CurrentThread.ManagedThreadId, jid);
	}

	public static void Main(string[] args)
	{
		int n = args.Length > 0 ? int.Parse(args[0]) : 50;
		
		//HandleJob(n);
		Thread child = new Thread(delegate(){HandleJob(n);});
		child.IsBackground = n > 75;
		child.Start();
		
		HandleJob(40);
	}
}

File: C/UNIX/Concurrency/threadtest.c
#include "worker.h"
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <pthread.h>

void HandleJob(int jid)
{
	printf("Thread <%x> has accepted job:%d\n", pthread_self(), jid);
	DoWork(jid);
	printf("Thread <%x> has finished job:%d\n", pthread_self(), jid);
}

void* ChildStart(void* arg)
{
	HandleJob(6);
}

int main(void)
{
	pthread_t child;

	pthread_create(&child, NULL, ChildStart, NULL);

	HandleJob(5);

	pthread_join(child, NULL);
}















